lib/cequel/record/record_set.rb

```diff
@@ -846,15 +846,16 @@ def filter_secondary_index(column_name, value)
                "Can't scope key column #{column_name} without also scoping " \
                "#{missing_column_names.join(', ')}"
         end
-        if scoped_indexed_column && !allow_filtering
+        if scoped_secondary_columns && !allow_filtering
           fail IllegalQuery,
-               "Can't scope by more than one indexed column in the same query"
+               "Can't scope by more than one indexed column in the same query without allow_filtering!"
         end
         unless column.indexed? || allow_filtering
           fail ArgumentError,
-               "Can't scope by non-indexed column #{column_name}"
+               "Can't scope by non-indexed column #{column_name} without allow_filtering!"
         end
-        scoped(scoped_indexed_column: {column_name => column.cast(value)})
+        filter = { column_name => column.cast(value) }
+        scoped(scoped_secondary_columns: scoped_secondary_columns ? scoped_secondary_columns.merge(filter) : filter)
```

Comment:
The logic in the method is really complicated. what if we reorganized it like

```
       if allow_filtering
         # bypass safety checks

       elsif scoped_secondary_columns.any?
          fail IllegalQuery,
               "Can't scope by more than one indexed column in the same query without allow_filtering!"

        elsif ! column.indexed?
          fail ArgumentError,
               "Can't scope by non-indexed column #{column_name} without allow_filtering!"
        end

        scoped(scoped_secondary_columns: 
               scoped_secondary_columns.merge(column_name => column.cast(value)) )
```

We would need to provide default value of `{}` for `scoped_secondary_columns` but i think that is better anyway. It allows the rest of the code, like this, to be less timid.

Reply:
@pezra I am not sure how to set the default value for `scoped_secondary_columns`, where should I look?

Reply:
Does sth like this look ok? I'm just missing the default value for `scoped_secondary_columns`

```ruby
def filter_secondary_column(column_name, value)
  column = target_class.reflect_on_column(column_name)
  validate_secondary_column_filter(column)
  scoped(scoped_secondary_columns:
    scoped_secondary_columns.merge(column_name => column.cast(value)))
end

def validate_secondary_column_filter(column)
  if column.nil?
    fail ArgumentError,
      "No column #{column_name} configured for #{target_class.name}"
  end
  if column.key?
    missing_column_names = unscoped_key_names.take_while do |key_name|
      key_name != column_name
    end
    fail IllegalQuery,
      "Can't scope key column #{column_name} without also scoping " \
      "#{missing_column_names.join(', ')}"
  else
    validate_non_key_column_filter(column)
  end
end

def validate_non_key_column_filter(column)
  return if allow_filtering

  if scoped_secondary_columns.any?
    fail IllegalQuery,
      "Can't scope by more than one indexed column in the same query without allow_filtering!"
  end
  unless column.indexed?
    fail ArgumentError,
      "Can't scope by non-indexed column #{column_name} without allow_filtering!"
  end
 end

Reply:
@pezra Nevermind! Found it! Code updated!

Reply:
@pezra Hello! Any chance you could look at the updated PR, please?